### ISSUE TITLE
AnimAssetManager with scaleFactor

### ISSUE THREAD

**Issue by ll24llll** [Gamua/Starling-Extension-Adobe-Animate#2]
Right now AnimAssetManager does not receive the scaleFactor parameter. How could it be managed?

**Comment by PrimaryFeather** [comment]
You're right, it seems I forgot to include the scale factor in the constructor, and to pass it on to the base class. I'm currently on vacation, but I can do that on Monday!

In the meantime, simply assign your custom scale factor to  `animAssetManager.textureOptions.scale`. That should do it just as well!

**Comment by ll24llll** [comment]
Dear Daniel,
I tried it but textures and animations are messed up.
```
        var assets:AnimAssetManager = new AnimAssetManager();
        assets.textureOptions.scale = contentScale;
```

Without setting contentScale animations are fine but everything has wrong size.

**Comment by PrimaryFeather** [comment]
Hm ... well, when Animate CC creates those textures, it creates them with a scale factor of one — and unfortunately, it doesn't allow any tweaking of this factor. A scale factor of two would mean, for example, that the individual element-textures would be twice as wide and high. 

That's why passing a scale factor different to "1" will lead to problems. Maybe that's why I didn't add that constructor argument. 

Actually, we'd need to finally get Adobe to add such an option. Perhaps you can start such a thread in the Adobe forums or ask support — I did that already in the past, but for now, without success. 😞

**Comment by ll24llll** [comment]
There are some threads related to this topic on adobe forums, [here ](https://forums.adobe.com/thread/2504988)and [here](https://forums.adobe.com/thread/2461946), they have not developed this feature yet. By the way thanks for your support. I try Dragon Bones for now, but it seems I have to create all animations inside Dragon Bones Pro again :-(

**Comment by PrimaryFeather** [comment]
Yeah, having to recreate all those animations with Dragon Bones would be a huge effort, I can understand the severity of this issue. :-(

I just created a quick set of instructions how we can make Adobe aware of the limitations of this feature, and get them to implement the fixes. It would be great if you could follow those steps! See: #3

**Comment by ll24llll** [comment]
I'm done with filling the adobe feature request forms. Hope we can get them to improve this feature. Thanks for your effort, nice job.

**Comment by PrimaryFeather** [comment]
Thanks for filling out the forms! Hopefully, we can get Adobe to acknowledge the lack of those features, soon. After all, none of them should be very hard to implement!
